### PR TITLE
Delegate transaction refactor

### PR DESCRIPTION
**Pull request by rizip1**
* Remove actions with "Old" postfix from delegation flow
* Remove race-condition check which is no longer necessary
* Move "deregisterStakeKey" to "delegate" file, instead of "poolOwner" file

For easier understanding, I suggest to review commit by commit.

### CODE REVIEW

**Review by xdzurman — APPROVED** [review]
LGTM

File: app/frontend/actions/delegate.ts [hunk 1]
```diff
@@ -131,10 +144,71 @@ export default (store: Store) => {
...
 
+  const delegate = async (state: State): Promise<void> => {
+    return await confirmTransaction(state, {
+      txConfirmType: TxType.DELEGATE,
+      txPlan: state.cachedTransactionSummaries[TxType.DELEGATE].plan,
+      sourceAccountIndex: state.sourceAccountIndex,
+    })
+  }
+
+  const deregisterStakingKey = async (state: State): Promise<void> => {
```
Comment (refi93) [inline app/frontend/actions/delegate.ts:155]: I find it a bit odd that the file grouping these actions is called delegate - if it tries to group actions related to staking, perhaps that would be a more descriptive name, but I understand that would trigger quite a lot of changes, so this is rather a note for a later iteration of refactoring

File: app/frontend/actions/delegate.ts [hunk 2]
```diff
@@ -131,10 +144,71 @@ export default (store: Store) => {
...
+    const supportError = getWallet().ensureFeatureIsSupported(CryptoProviderFeature.WITHDRAWAL)
```
Comment (refi93) [inline app/frontend/actions/delegate.ts:156]: ensureFeatureIsSupported seems to not return anything as per the typing: https://github.com/vacuumlabs/adalite/blob/e4cde52738dbb2e224f1d6df68a13a3113c12446/app/frontend/types.ts#L44

Comment (refi93) [reply app/frontend/actions/delegate.ts:156]: ah I see, I should be looking at the function with the same name, but declared in shelley-wallet.ts :facepalm: - this is madness :D If it returns a boolean, something like isFeatureSupported would imho be more spot-on as a name

File: app/frontend/components/pages/delegations/shelleyBalances.tsx [hunk 3]
```diff
@@ -66,11 +65,7 @@ const shelleyBalances = ({
         )}
       </div>
       {!!rewardsAccountBalance && (
-        <button
-          disabled={calculatingDelegationFee}
-          className="button secondary balance withdraw"
-          onClick={withdrawRewards}
-        >
+        <button className="button secondary balance withdraw" onClick={withdrawRewards}>
```
Comment (refi93) [inline app/frontend/components/pages/delegations/shelleyBalances.tsx:68]: didn't the disabling of the button while delegation fees are being computed serve as a protection against repeated triggering of the logic? Shouldn't we debounce it then?

Comment (rizip1) [reply app/frontend/components/pages/delegations/shelleyBalances.tsx:68]: Well, the `calculatingDelegationFee` is not related to `withdrawRewards`, but to calculation of delegation fees trigers immediately when user lands on this screen. Now they are not sharing transaction plans, so this is why it was removed. Do you see there some other potential problems?

Comment (rizip1) [reply app/frontend/components/pages/delegations/shelleyBalances.tsx:68]: And the repeated protection is not there, but also was not there, as `calculatingDelegationFee` is not set when `withdrawRewards` is called.

File: app/frontend/actions/delegate.ts (hunk 1 above)
Comment (rizip1) [reply app/frontend/actions/delegate.ts:155]: Thanks, I could create an issue for that, so it can be changed in standalone PR.

**Review by refi93 — APPROVED** [review]

**Pull request merged**